### PR TITLE
Stats docs

### LINKED ISSUE #631: Improve effective sample size docs
The [docs for `effective_sample_size`](https://arviz-devs.github.io/arviz/generated/arviz.effective_sample_size.html#arviz.effective_sample_size) have some typos and unexplained notation. 

This issue is a remainder to correct them once #623 is merged.

### PR DESCRIPTION
**Pull request by OriolAbril**
Minor fixes on stats module docs and adding examples. Fixes #631

### CODE REVIEW

**Comment by canyon289** [comment]
LGTM

**Comment by ahartikainen** [comment]
These are annoying

```
************* Module arviz.stats.diagnostics
arviz/stats/diagnostics.py:120:0: C0301: Line too long (102/100) (line-too-long)
arviz/stats/diagnostics.py:182:0: C0301: Line too long (102/100) (line-too-long)
************* Module arviz.stats.stats
arviz/stats/stats.py:382:0: C0301: Line too long (106/100) (line-too-long)
```

**Comment by OriolAbril** [comment]
I will probably finish around Monday, having included simple examples for all functions in stats and diagnostics.

**Comment by OriolAbril** [comment]
Ready for review.

File: arviz/stats/diagnostics.py [hunk 1]
```diff
@@ -139,36 +154,62 @@ def ess(data, *, var_names=None, method="bulk", relative=False, prob=None):
...
-        Return the effective sample size for mean, :math:`\hat{N}_{eff}`
+        Return the effective sample size, :math:`\hat{N}_{eff}`
 
     Notes
     -----
     The basic ess diagnostic is computed by:
 
     .. math:: \hat{N}_{eff} = \frac{MN}{\hat{\tau}}
-    .. math:: \hat{\tau} = -1 + 2 \sum_{t'=0}^K \hat{P}_t'
+    .. math:: \hat{\tau} = -1 + 2 \sum_{t'=0}^K \hat{P}_{t'}
```
Comment (ColCarroll) [inline arviz/stats/diagnostics.py]: nice catch

File: arviz/stats/diagnostics.py [hunk 2]
```diff
@@ -311,7 +370,7 @@ def rhat(data, *, var_names=None, method="rank"):
 
 
 def mcse(data, *, var_names=None, method="mean", prob=None):
-    """Calculate Markov Chain Standard Error for statistic.
+    """Calculate Markov Chain Standard Error statistic.
```
Comment (ColCarroll) [inline arviz/stats/diagnostics.py]: it'd be cool to have a reference for this eventually (but certainly not needed in this PR)

File: arviz/stats/stats.py [hunk 3]
```diff
@@ -29,9 +29,9 @@ def compare(
     seed=None,
     scale="deviance",
 ):
-    r"""Compare models based on WAIC or LOO cross validation.
+    r"""Compare models based on WAIC or LOO cross-validation.
 
-    WAIC is Widely applicable information criterion, and LOO is leave-one-out
+    WAIC is the Widely applicable information criterion, and LOO is leave-one-out
```
Comment (ColCarroll) [inline arviz/stats/stats.py]:
```suggestion
    WAIC is the widely applicable information criterion, and LOO is leave-one-out
```

File: arviz/stats/stats.py [hunk 4]
```diff
@@ -643,29 +688,37 @@ def summary(
...
+           ...:     sd = np.sqrt(np.mean((x-median)**2))
+           ...:     return sd
+           ...:
+           ...: func_dict = {
+           ...:     "std": np.std,
+           ...:     "median_std": median_sd,
+           ...:     "5%": lambda x: np.percentile(x, 5),
+           ...:     "median": lambda x: np.percentile(x, 50),
+           ...:     "95%": lambda x: np.percentile(x, 95),
+           ...: }
```
Comment (ColCarroll) [inline arviz/stats/stats.py]: this is nice

File: arviz/stats/diagnostics.py [hunk 5]
```diff
@@ -196,13 +202,13 @@ def ess(data, *, var_names=None, method="bulk", relative=False, prob=None):
 
     .. ipython::
 
-        In [2]: az.ess(data, relative=True, var_names=["mu", "theta_t"])
+        In [1]: az.ess(data, relative=True, var_names=["mu", "theta_t"])
 
     Calculate the ess using the "tail" method, which requires the `prob` argument
```
[the remark this answers is not in the document]
Comment (OriolAbril) [reply arviz/stats/diagnostics.py]: Do you prefer to remove it or to add a note advising against it?

[the remark this answers is not in the document]
Comment (ahartikainen) [reply arviz/stats/diagnostics.py]: Maybe remove it. I added it as an optional so one can do symmetrical minimum for different quantiles. Or maybe add a comment, but don't provide it in the example.

So if users thinks prob=0.1 is better then tail calculates symmetrical tail with that (min(ess_0.1, ess_1-0.1))

**Comment by ColCarroll** [comment]
Thanks, @OriolAbril !

**Comment by ahartikainen** [comment]
Hi, sorry to comment too late, but yes I think the `tail` method should be there, just the prob should not be touched? No need to reopen this PR, maybe quick fix with another one?

**Comment by OriolAbril** [comment]
> Hi, sorry to comment too late, but yes I think the `tail` method should be there, just the prob should not be touched? No need to reopen this PR, maybe quick fix with another one?

Oh, sorry, I missunderstood. I will fix it in #678. I wanted to merge this before getting into more lengthy work so that docs are updated as soon as possible.

**Pull request merged**